user-operator/src/test/java/io/strimzi/operator/user/operator/KafkaUserQuotasTest.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.user.operator;
+
+import io.strimzi.test.EmbeddedZooKeeper;
+import io.vertx.core.json.JsonObject;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class KafkaUserQuotasTest {
+
+    private static EmbeddedZooKeeper zkServer;
+
+    private KafkaUserQuotas kuq;
+
+    private JsonObject quotasJson;
+
+
+    @BeforeAll
+    public static void startZk() throws IOException, InterruptedException {
+        zkServer = new EmbeddedZooKeeper();
+    }
+
+    @AfterAll
+    public static void stopZk() {
+        zkServer.close();
+    }
+
+    @BeforeEach
+    public void createKUQ() {
+        quotasJson = new JsonObject();
+        quotasJson.put("consumer_byte_rate", "1000");
+        quotasJson.put("producer_byte_rate", "2000");
+        kuq = new KafkaUserQuotas(zkServer.getZkConnectString(), 6_000);
+    }
+
+    @Test
+    public void normalCreate() {
+        JsonObject quotasJson = new JsonObject();
+        quotasJson.put("consumer_byte_rate", "1000");
+        quotasJson.put("producer_byte_rate", "2000");
+        kuq.createOrUpdate("normalCreate", quotasJson);
+    }
+
+    @Test
+    public void doubleCreate() {
+        kuq.createOrUpdate("doubleCreate", quotasJson);
+        kuq.createOrUpdate("doubleCreate", quotasJson);
+    }
+
+    @Test
+    public void normalDelete() {
+        kuq.createOrUpdate("normalDelete", quotasJson);
+        kuq.delete("normalDelete");
+    }
+
+    @Test
+    public void doubleDelete() {
+        kuq.createOrUpdate("doubleDelete", quotasJson);
+        kuq.delete("doubleDelete");
```

Comment:
Check the quotas exist after the create and don't exist after the delete. And notification path.